amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/declaration/external/raml/RamlJsonSchemaExpression.scala

```diff
@@ -174,9 +176,9 @@ case class RamlJsonSchemaExpression(key: YNode,
                              value: YNode,
                              extLocation: Option[String]): AnyShape = {
 
-    val url               = extLocation.flatMap(ctx.declarations.fragments.get).flatMap(_.location)
-    val parser            = getParser(text, valueAST, url)
-    val schemaEntry       = YMapEntry(key, parser.document().node)
+    val parser            = jsonParser(extLocation, text, valueAST)
```

Comment:
Why build the parser before checking if the node already exists?

Reply:
[OOF] - It can also be more readable to explicitly state in this function that you will either get the already parsed node or parse it. Delegating it to the `Helper` may be obscuring it too much.